meshnets/modules/decoder.py

```diff
@@ -1,10 +1,29 @@
 """"Define the GraphDecoder class."""
 
+import torch
+from torch.nn import Identity
+from torch_geometric.data import Batch
 
-class GraphDecoder:
 
-    def __init__(self, *args):
-        pass
+class GraphDecoder(torch.nn.Module):
+    """Decoder for graphs.
+    
+    Decode latent graphs to an output Tensor corresponding to predicted
+    node features by applying an MLP decoder to each node features.
+    """
 
-    def __call__(self, *args):
-        pass
+    def __init__(self, output_size, latent_size, num_mlp_layers):
+        """Initialize the node features decoder given the output size,
+        latent size and number of MLP layers."""
+        super().__init__()
+
+        # TODO(victor) : implement decoder
+        self.node_decoder = Identity(output_size, latent_size, num_mlp_layers)
+
+    def forward(self, graph: Batch) -> torch.Tensor:
```

Comment:
I am assuming that the decoder will appear after several layers of other messaging passing rounds. Therefore, since the output of [most layers](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/simple_conv.html#SimpleConv.forward) seems to be tuples. We should have this receive `node_atributes, edge_index` or something. But I am still not sure exactly what the output should be. This will be something to find out as we go along. Therefore we can leave this as is and refactor in the future if needed